sdk/tables/azure-tables/src/serialization.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { EdmGuid } from "./EdmGuid";
+import { EdmInt64 } from "./EdmInt64";
+import { isUint8Array, encodeByteArray, decodeString } from "./utils/bufferSerializer";
+
+const Edm = {
+  DateTime: "Edm.DateTime",
+  Int64: "Edm.Int64",
+  Guid: "Edm.Guid",
+  Binary: "Edm.Binary"
+};
+
+export function serialize(obj: any): object {
+  const serialized: any = {};
+  for (const [key, value] of Object.entries(obj)) {
+    if (value instanceof Date) {
+      serialized[key] = value;
+      serialized[`${key}@odata.type`] = Edm.DateTime;
+    } else if (value instanceof EdmInt64) {
+      serialized[key] = value.value;
+      serialized[`${key}@odata.type`] = Edm.Int64;
+    } else if (value instanceof EdmGuid) {
+      serialized[key] = value.value;
+      serialized[`${key}@odata.type`] = Edm.Guid;
+    } else if (isUint8Array(value)) {
+      serialized[key] = encodeByteArray(value as Uint8Array);
+      serialized[`${key}@odata.type`] = Edm.Binary;
+    } else {
+      serialized[key] = value;
+    }
+  }
+  return serialized;
+}
+
+function getTypedObject(value: any, type: string): any {
+  switch (type) {
+    case Edm.DateTime:
+      return new Date(value);
+    case Edm.Int64:
+      return new EdmInt64(value);
+    case Edm.Guid:
+      return new EdmGuid(value);
+    case Edm.Binary:
+      return decodeString(value);
+    default:
+      return value;
+  }
+}
+
+export function deserialize(obj?: any): object {
```

Comment:
maybe it would be useful for `deserialize` to be a generic so you could provide the actual return type if you know it?

Reply:
Yes, I will be adding generics to this function and some other methods in a followup PR